test/ResponseTest.php

```diff
@@ -112,6 +112,13 @@ public function testCanSetCustomReasonPhrase()
         $this->assertSame('Foo Bar!', $response->getReasonPhrase());
     }
 
+    public function testCannotCreateReasonPhraseNotString()
```

Comment:
Let's rephrase this to `testWithStatusRaisesAnExceptionForNonStringReasonPhrases()`. Additionally, please test it with other non-string values as well (using a data provider), to validate they all produce an exception.